api/topology/v1/topology_api.proto

```diff
@@ -31,6 +40,57 @@ message GetTopologyResponse {
   repeated QueryResult results = 1;
 }
 
+message SearchTopologyRequest {
+  message FieldSelector {
+    enum Column {
+      UNSPECIFIED = 0;
+      ID = 1;
+    }
```

Comment:
and maybe this could just be a general field mask for selector, so `metadata.level1.fieldname` or `id` at the top level rather than differentiating. just a thought, you probably have thought more of implications here.

Reply:
Ah yeah true i was mostly trying to work around the field mask selector in this case, an easy way to delineate columns vs metadata. I like the field mask approach, let me give that a try.